packages/handler-next/src/createNextSynthqlHandler.ts

```diff
@@ -123,21 +112,26 @@ async function tryParseRequest(req: NextSynthqlHandlerRequest) {
 
 async function tryExecuteQuery<DB>(
     queryEngine: QueryEngine<DB>,
-    query: any,
-    returnLastOnly: boolean,
-) {
-    return queryEngine.execute(query, { returnLastOnly });
-}
-
-async function tryExecuteRegisteredQuery<DB>(
-    queryEngine: QueryEngine<DB>,
-    { queryId, params }: { queryId: string; params: Record<string, unknown> },
+    queryOrBody: any,
     returnLastOnly: boolean,
 ) {
-    return queryEngine.executeRegisteredQuery(
-        { queryId, params },
-        { returnLastOnly },
-    );
+    if (isRegisteredQueryRequest(queryOrBody)) {
+        return queryEngine.executeRegisteredQuery(
+            {
+                queryId: queryOrBody.queryId,
+                params: queryOrBody.params,
+            },
+            {
+                returnLastOnly,
+            },
+        );
+    } else if (isRegularQueryRequest(queryOrBody)) {
+        return queryEngine.execute(queryOrBody.query as Query<DB>, {
```

Comment:
this cast shouldn't be needed: you should move the cast into `isRegularQueryRequest`